server/src/test/java/io/crate/integrationtests/InsertIntoIntegrationTest.java

```diff
@@ -1993,4 +1995,36 @@ ON CONFLICT (real_id, \"date\", year)
             "2030| 99998.0"
         );
     }
+
+    @Test
+    public void test_non_deterministic_sub_column_not_included_in_target_cols() {
+        execute("""
+            CREATE TABLE tbl (
+                a int PRIMARY KEY,
+                o1 object as (
+                    sub int as round((random() + 1) * 100)
+                )
+            )
+            """
+        );
+
+        execute("INSERT INTO tbl(a) VALUES (1)");
+        refresh();
+
+        // Ensure that the same value is stored on primary and replica.
+        List<Integer> results = new ArrayList<>();
+        for (Sessions sqlOperations : cluster().getDataNodeInstances(Sessions.class)) {
```

Comment:
Maybe you can iterate over the nodes, and use directly this variant of execute:
```
execute(String stmt, Object[] args, String node)
```

Reply:
I tried `cluster().getNodeNames()` + using those as param for `execute` but it had 5 nodes 2 data nodes + 3 master eligible nodes. And there is no code to filter out non-data nodes so I took another approach, similar to what we do in `testJobLogWithEnabledAndDisabledStats`

Reply:
There was an issue with not using a session, created a follow up PR https://github.com/crate/crate/pull/15275

Reply:
thx!